runtime/fastly/builtins/kv-store.cpp

```diff
@@ -474,6 +475,24 @@ bool KVStore::put(JSContext *cx, unsigned argc, JS::Value *vp) {
       ttl = parsed;
     }
 
+    JS::RootedValue gen_val(cx);
+    if (!JS_GetProperty(cx, opts_val, "gen", &gen_val)) {
+      return false;
+    }
+
```

Comment:
the generation match integer comes across as a uint64_t, however I'm not converting it to [uint64](https://github.com/bytecodealliance/StarlingMonkey/blob/main/builtins/web/file.cpp#L16).  I noticed some functions in spidermonkey that had a comment of it being non-standard, could/should I be using those?

Reply:
Adding the support for BigInt can be useful for 64 bit integers effectively supporting `gen: 1000n` as well as `gen: 1000` say. The BigInt functions are exactly what you want to use and there should be existing examples in the codebase for this as well.

Reply:
I couldn't find a good example of converting a `RootedBigInt` and converting that to a uint64_t, so I left it as a `number`